src/script_interface/packed_variant.hpp

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) 2020 The ESPResSo project
+ *
+ * This file is part of ESPResSo.
+ *
+ * ESPResSo is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ESPResSo is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef SCRIPT_INTERFACE_PACKED_VARIANT_HPP
+#define SCRIPT_INTERFACE_PACKED_VARIANT_HPP
+
+#include "Variant.hpp"
+
+#include <functional>
+#include <unordered_map>
+#include <utility>
+
+namespace ScriptInterface {
+using ObjectId = std::size_t;
+
+/**
+ * @brief Id for object.
+ *
+ * This assigns every ObjectHandle a unique id.
+ */
+inline ObjectId object_id(const ObjectHandle *p) {
+  static_assert(sizeof(const ObjectHandle *) <= sizeof(ObjectId), "");
+  return reinterpret_cast<ObjectId>(p);
+}
```

Comment:
This code is only used on the head node as a hash function, so this should be fine. The polymorphism doesn't matter because the address as a `ObjectHandle` is unique within every object. It will not be at the start of the object for derived
types, but this does not matter for this purpose.